addons/api/addon/models/role.js

```diff
@@ -42,6 +51,61 @@ export default class RoleModel extends GeneratedRoleModel {
       : null;
   }
 
+  /**
+   * Convenience for looking up the grant scopes, if loaded.
+   */
+  get grantScopes() {
+    // filter out scopes that start with an o_ or p_
+    // then sort the keywords
+    const grantScopes = [];
+    if (this.grant_scope_ids) {
+      const keywordIDs = this.grant_scope_ids.filter((id) => {
+        return !id.startsWith('o_') && !id.startsWith('p_') && id !== 'global';
+      });
+      // using filter instead of find to get the global id and allow for
+      // spreading into sortedScopeIDs if globalID is undefined
+      const globalID = this.grant_scope_ids.filter((id) => {
+        return id === 'global';
+      });
+      const orgIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('o_');
+      });
+      const projectIDs = this.grant_scope_ids.filter((id) => {
+        return id.startsWith('p_');
+      });
+
+      const sortedKeywords = keywordIDs.sort((a, b) => {
+        return (
+          GRANT_SCOPE_KEYWORDS.indexOf(a) - GRANT_SCOPE_KEYWORDS.indexOf(b)
+        );
+      });
+
+      const sortedScopeIDs = [
+        ...sortedKeywords,
+        ...globalID,
+        ...orgIDs,
+        ...projectIDs,
+      ];
```

Comment:
If the goal was for a custom sort order, why not have just written a custom sort function for all of these in the first place instead of just for the `sortedKeywords`?

Reply:
I actually did have a custom sort order originally but I felt it was overkill just to sort the keywords in the correct order along with orgs and scope. I decided to go with filtering it down. I don't feel like this will be a performance issue but happy to talk some more about it.

Reply:
Why is it overkill when you're just doing the same thing in a more manual way? Doesn't it simplify the logic greatly?

Reply:
Resolving this comment and will continue in new comment.